### PR TITLE
style: reformat all TOML files with `taplo`

### PR DESCRIPTION
**Pull request by rami3l**
As [discussed and approved](https://discord.com/channels/442252698964721669/463480252723888159/1261135626879766619) by @rbtcollins.

## Prior Art
- https://github.com/rust-lang/team/pull/1338

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (rami3l) [inline .github/workflows/ci.yaml]: This is to make it clearer that `git diff` actually checks for coherence, and will actually print out the inconsistencies if there are any.

File: Cargo.toml
Comment (djc) [inline Cargo.toml]: So one reason I've bounced off of this in the past is that IMO this is not an improvement. I think one line per dependency/features is a pretty strong default and having just a few dependencies that spread their features out over multiple lines does not make the `Cargo.toml` easier to read.

Comment (rami3l) [reply Cargo.toml]: > So one reason I've bounced off of this in the past is that IMO this is not an improvement. I think one line per dependency/features is a pretty strong default and having just a few dependencies that spread their features out over multiple lines does not make the `Cargo.toml` easier to read.

@djc Can we make max line lengths extra large, like 150?

Comment (rami3l) [reply Cargo.toml]: @djc In other words I don't care that much about the style per se, but rather that the formatting can be automated...

File: .github/workflows/ci.yaml
Comment (smoelius) [inline .github/workflows/ci.yaml]:
```suggestion
          git diff --exit-code
```
Sorry for the intrusion. `--quiet` [implies `--exit code`](https://git-scm.com/docs/git-diff). Plain `git diff` will exit with 0 even if there are differences.

Comment (rami3l) [reply .github/workflows/ci.yaml]: @smoelius Thanks a lot!

Comment (djc) [reply .github/workflows/ci.yaml]: @smoelius no intrusion at all, always happy to get your feedback!

File: Cargo.toml
Comment (djc) [reply Cargo.toml]: > > So one reason I've bounced off of this in the past is that IMO this is not an improvement. I think one line per dependency/features is a pretty strong default and having just a few dependencies that spread their features out over multiple lines does not make the `Cargo.toml` easier to read.
> 
> @djc Can we make max line lengths extra large, like 150?

Let's try it?

Comment (rami3l) [reply Cargo.toml]: @djc [Special rule](https://taplo.tamasfe.dev/configuration/file.html#rules) applied: `column_width = 160` for `dependencies` exclusively.

Comment (rami3l) [reply Cargo.toml]: @djc Sorry, I've forgot `workspace.`, `dev-` and `build-dependencies`, just fixed.

File: .github/workflows/ci.yaml
Comment (rami3l) [inline .github/workflows/ci.yaml:1398]: The latest version is 0.9, but for now both Neovim/mason and VSCode are on 0.8, so I'll stick with that until later.

Comment (rami3l) [reply .github/workflows/ci.yaml:1398]: Neovim/mason now has `taplo` 0.9 (https://github.com/mason-org/mason-registry/pull/6469), but I guess the PR will make more sense once the VSCode plugin has been updated to 0.9.

This might take a while since they'll have to wait for the maintainer to rotate the access token to the marketplace (https://github.com/tamasfe/taplo/issues/624)... ~~Marking as draft until this problem is resolved.~~

**Review by djc — APPROVED** [review]
Nice!

Suggest putting the reformatting commit before the one that adds the CI check.

**Comment by rami3l** [comment]
> Suggest putting the reformatting commit before the one that adds the CI check.

@djc I've intentionally put the reformatting commit at the end so I can amend to it conveniently, but fair enough.

**Comment by djc** [comment]
> > Suggest putting the reformatting commit before the one that adds the CI check.
> 
> @djc I've intentionally put the reformatting commit at the end so I can amend to it conveniently, but fair enough.

Makes sense -- can always reorder before the merge?

**Comment by rami3l** [comment]
> Makes sense -- can always reorder before the merge?

@djc Yes, but that's a draft, so... Anyway, we are on the same page now.

**Comment by rami3l** [comment]
@djc As there are a bit too many regressions/breakages in `taplo` v0.9, I've made the upgrade a separate follow-up issue in https://github.com/rust-lang/rustup/issues/3945. So far we are still using v0.8 for our CI.

**Pull request merged**
